### DISCUSSION TITLE
Why no cloud buffer for cirrus?

### DISCUSSION THREAD

**Discussion by maxfreu** [davidfrantz/force#224] [General]
Hi! It seems like cirrus clouds are not buffered. Is this intentional? If yes, why? Here's an example where I would think a buffer could be useful to exclude compromised pixels:
![image](https://user-images.githubusercontent.com/67329240/190382720-5342cc85-f101-42fc-9fa0-e586c5d60212.png)

**Comment by davidfrantz** [comment]
Hi,

this was a compromise decision at some point. Cirrus clouds are quite hard to detect reliably (you can read on accuracies and difficulties in the [CMIX](https://doi.org/10.1016/j.rse.2022.112990). Accuracies significantly drop down once cirrus clouds are considered in the validation (applies to all processors btw).

Buffering cirrus clouds often results in flagging out huge areas. To avoid that, we somewhen took the decision to not go for a cirrus buffer. 

But yes, I guess that at least an option altogether with the other buffering options might make sense though. This should not be too hard to implement and I will put it on the to-do heap. Thanks for bringing this up again.

Cheers,
David

**Comment by davidfrantz** [comment]
I experimented a bit with adding a cirrus buffer. As I cannot redefine the QAI bit structure without disruption, I see two options to encode the cirrus buffer, though I am undecided what the better option would be. 

1) buffered cirrus clouds are not flagged independently as buffered, i.e. buffer and cirrus detection are subsumed in the cirrus flag (like it is done with the snow buffer)
2) buffered cirrus clouds are added to the standard buffered cloud flag. This means that there is no computational distinction in the origin of the buffer and funny shapes might occur when filtering for opaque clouds, buffered clouds, but not cirrus clouds. However, buffered clouds can be left out altogether in an analysis.

Do you have an opinion on this?

**Reply by maxfreu** [reply]
I think option 1 makes more sense. Then one can select whether or not to get rid of cirrus clouds completely.

**Comment by davidfrantz** [comment]
Hi @maxfreu 

An optional cirrus buffer is now available within the dev branch: https://github.com/davidfrantz/force/commit/6f1335aa2e0c86e1a0dc41a4d29b138963f384b4

Excerpt from version history:

> It is now possible to enable a buffer by using a buffer size > 0 for the new parameter CIRRUS_BUFFER. Note, that both the cirrus buffer won’t be represented in the “cloud buffer” QAI bit, but will be subsumed in the cirrus cloud bit (similar to snow buffer). This is because the bit structure would need to be disruptively changed to accomodate for a distinction.

Cheers,
David

**Reply by maxfreu** [reply]
Thank you! I will try it out when I'm back from vacation :)
